executor/windowfunc/func_row_number.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package windowfuncs
+
+import (
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+type rowNumber struct {
+	baseWindowFunc
+}
+
+type partialResult4RowNumber struct {
+	curIdx int64
+}
+
+func (wf *rowNumber) ProcessOneChunk(sctx sessionctx.Context, rows []chunk.Row, pr PartialResult, dest *chunk.Chunk, remained int) ([]chunk.Row, int, error) {
+	p := (*partialResult4RowNumber)(pr)
+	for len(rows) > 0 && remained > 0 {
+		p.curIdx++
+		remained--
+		dest.AppendInt64(wf.ordinal, p.curIdx)
```

Comment:
Hmm, can we treat `row_number` as a special aggregate function? we can increase the row number in its `AppendFinalResult2Chunk`, and reset the row number to 1 in its `ResetPartialResult`, then we don't need to change the executor framework for window function at all?

Reply:
OK, got it.

Reply:
In window executor, we buffered all the rows belonging to the same partition even for `row_number()`, seems it is not streaming processed?

Reply:
No, for `row_number`, it buffers at most one chunk. The buffered rows are controlled by each function according to their properties.